### ISSUE TITLE
size or __len__ method

### ISSUE THREAD

**Issue by joezuntz** [LSSTDESC/gcr-catalogs#164]
We need catalog objects to provide a `__len__` method or `size` attribute.

Many uses, especially for large catalos, need to allocate space for things before running on the whole catalog.  This could even just be a hard-coded value for different catalogs.

I wrote this one specifically for the protoDC2 test catalog - I assume it won't work for others:

```
    def get_catalog_size(self, gc):
        """
        Get the size of a GCR catalog object.
        This is not robust - catalogs can be split over
        multiple files.  As of now there is no API for this.

        Parameters
        ----------

        gc: GCRCatalogs catalog object

        Returns
        -------
        n: integer
            Number of catalog rows
        """
        import h5py
        filename = gc.get_catalog_info()['filename']
        print(f"Reading catalog size directly from {filename}")
        # Take the name of a specific column we expect to exist
        f = h5py.File(filename)
        n = f['galaxyProperties/ra'].size
        f.close()
        return n

```

**Comment by yymao** [comment]
@joezuntz thanks for the suggestion. One question I have is how this method should behave for catalogs that have chunks (e.g., the coadd catalog). Should this method return the total number of entries (summing over all chunks) or should there be an option for this method to return an iterator that iterate over chunks and return the length for each chunk? 

Also, for some catalogs, different columns actually have different lengths. I think for those catalogs this method will just raise an error?

**Comment by yymao** [comment]
@joezuntz just following up on this to see if you can comment on which behavior (always just return total length or return an iterator of lengths that correspond to data chunks) would work better for your use case?

**Comment by joezuntz** [comment]
Sorry, forgot to get back to you.

Total length would be needed for my use case, though of course you could get that from just the sum of the other option.

**Comment by yymao** [comment]
Thanks, @joezuntz. This is implemented GCR 0.8.0 (will be available once we update the share environment)
